sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/cpp/CppGenerator.java

```diff
@@ -1121,51 +1121,83 @@ private CharSequence generateArrayProperty(
-            indent + "        return length;\n" +
-            indent + "    }\n\n",
-            toUpperFirstChar(propertyName),
-            token.arrayLength(),
-            generateArrayFieldNotPresentCondition(token.version(), indent),
-            offset,
-            cppTypeName));
-
-        sb.append(String.format(
-            indent + "    %1$s &put%2$s(const char *src)\n" +
-            indent + "    {\n" +
-            indent + "        std::memcpy(m_buffer + m_offset + %3$d, src, sizeof(%4$s) * %5$d);\n" +
-            indent + "        return *this;\n" +
-            indent + "    }\n\n",
-            containingClassName,
-            toUpperFirstChar(propertyName),
-            offset,
-            cppTypeName,
-            token.arrayLength()));
+        if (token.encoding().primitiveType() != PrimitiveType.CHAR)
+        {
+            sb.append(String.format(
+                indent + "    std::uint64_t get%1$s(char *dst, const std::uint64_t length) const\n" +
+                indent + "    {\n" +
+                indent + "        if (length > %2$d)\n" +
+                indent + "        {\n" +
+                indent + "             throw std::runtime_error(\"length too large for get%1$s [E106]\");\n" +
+                indent + "        }\n\n" +
+                    "%3$s" +
+                indent + "        std::memcpy(dst, m_buffer + m_offset + %4$d, sizeof(%5$s) * length);\n" +
+                indent + "        return length;\n" +
+                indent + "    }\n\n",
+                toUpperFirstChar(propertyName),
+                token.arrayLength(),
+                generateArrayFieldNotPresentCondition(token.version(), indent),
+                offset,
+                cppTypeName));
 
-        if (token.encoding().primitiveType() == PrimitiveType.CHAR)
+            sb.append(String.format(
+                indent + "    %1$s &put%2$s(const char *src) SBE_NOEXCEPT\n" +
+                indent + "    {\n" +
+                indent + "        std::memcpy(m_buffer + m_offset + %3$d, src, sizeof(%4$s) * %5$d);\n" +
+                indent + "        return *this;\n" +
+                indent + "    }\n\n",
+                containingClassName,
+                toUpperFirstChar(propertyName),
+                offset,
+                cppTypeName,
+                token.arrayLength()));
+        }
+        else
         {
             sb.append(String.format(
+                indent + "    std::uint64_t get%1$s(char *dst, std::uint64_t length) const SBE_NOEXCEPT\n" +
+                indent + "    {\n" +
+                indent + "        std::uint64_t minlen = std::min<std::uint64_t>(length, %3$d);\n" +
+                indent + "        const char* src = m_buffer + m_offset + %2$d;\n" +
+                indent + "        const char* end = sbe_util::stpncpy(dst, src, minlen);\n" +
+                indent + "        if (length > 6)\n" +
```

Comment:
Is the hard coded value of 6 here intentional?  If the intention is to null terminate dst when possible, I would expect that the compare would be against minlen.

Reply:
You're absolutely right. Well spotted! Pushing update now.

Reply:
Glad to help.  Getting this accepted would help overcome the issues described your Gist.